platforms/avr/config.h

```diff
@@ -0,0 +1,5 @@
+// Copyright 2023 Nick Brassel (@tzarc)
```

Comment:
Nothing currently adds this file to `CONFIG_H` and i guess it feels a bit odd having this functionality injected everywhere. If its just for our purposes, it feels more like and extension of the current `quantum/util.h`?

So maybe we could add `platforms/<plat>/_util.h` and a 
```
#if __has_include_next("_util.h")
#    include_next "_util.h" /* Include the platforms _util.h */
#endif
```

(And better name than `_util` maybe?)

Reply:
> We generally follow the pattern of xxxx.h -> platforms/<platform>/_xxxx.h don't we?

Ish, as it somewhat removes ambiguity in include order. Some of the older ones still need renaming.